ivadomed/utils.py

```diff
@@ -688,28 +690,17 @@ def segment_volume(folder_model, fname_image, fname_roi=None):
         with torch.no_grad():
             preds = model(batch['input'])
 
-        rdict = {}
-        rdict['gt'] = preds
-        batch.update(rdict)
-        # Take only metadata from one input
-        batch["input_metadata"] = batch["input_metadata"][0]
-
         # Reconstruct 3D object
-        for i_slice in range(len(batch['gt'])):
-            # Undo transformations
-            rdict = {}
-            # Import transformations parameters
-            for k in batch.keys():
-                if len(batch[k]):
-                    rdict[k] = batch[k][i_slice]
-            rdict_undo = undo_transforms(rdict)
+        for i_slice in range(len(preds)):
+            # undo transformations
```

Comment:
stupid question here: why do we need to undo transformations during inference? also, how come data_type='gt' (we don't have ground truth during normal use-case inference)

Reply:
Transformations are undone so that the prediction mask match the original image. For instance, if I crop the image 48x48 during inference, but my original image is 100x100, I want the prediction to superpose the 100x100 image. 

For the data_type='gt', the transforms are organized in a dictionary with the keys 'gt' (segmentation), 'img' (input image) and 'roi' (roi to help crop the image around a mask in 2D) (e.i. `transforms = {'gt': transforms_applied_to_gt, 'img':  transforms_applied_to_img, 'roi': transforms_applied_to_roi}`). For example. we don't want to normalize the gt (or any segmentation), so NormalizeInstance will only be in the key 'img'. This was added to remove all the [if calls](https://github.com/neuropoly/ivado-medical-imaging/blob/master/ivadomed/transforms.py#L1055) in the transforms (if self.labeled: do thing for labels) and the code duplication to apply the transform to the image then to the gt. So here 'gt' refers to the type of input that is given -> we only want to apply the 'gt' transforms since predictions are processed the same way as 'gt'. The name 'gt' may not be the more adapted, we could change it to 'seg' or 'segmentation' or 'mask'?

Reply:
cool! thank you for the explanation. 

> we could change it to 'seg'

👍  i would also suggest the more general 'label' (which can include segmentations, single points, etc.)

Reply:
The reason for not choosing `seg` is that the `roi` is also a segmentation.
In the config files we opted for `target`.
But I agree that, at the end, this data is the segmentation (target or predicted) --> so `seg` could make more sense. To discuss :)

Reply:
so why is there a 'roi' and a 'gt' (or 'seg') case? are there treatment exceptions between the two? (i guess there is, otherwise you would have called it the same thing)? or to rephrase, should there be different treatments?

Reply:
Yes `roi` and `gt` have not the same treatment --> we use the roi only for the crop --> passed this transform we do not use it anymore.
The requirements are not the same: Only need 1 ROI, but we may need several GT in case of multi label.
Also, ROI is optional, while GT is obviously mandatory.
So in brief: there are different treatments between the two.

Reply:
ah, makes sense. Is this documented somewhere?

Reply:
The difference between ROI and GT is explained in Wiki / config file (cf `*_suffix`): https://github.com/neuropoly/ivado-medical-imaging/wiki/configuration-file